### ISSUE TITLE
[open source] Bring back labels when checking PR titles

### ISSUE THREAD

**Issue by AquiGorka** [Lilypad-Tech/lilypad#245]
We would like to label PRs automatically in a GitHub action based on the conventional commit in the PR title. 

We had this functionality before through the [PRConventionalCommits](https://github.com/ytanikin/PRConventionalCommits) GitHub action, but the PRConventionalCommits would remove existing labels that are needed for our release pipeline.

It would be great to have these labels back! We want the labels to be appended without overwriting existing labels. PRConventionalCommits started some work to append instead of overwrite, but it doesn't appear complete yet.

We could add a custom GtiHub action to apply the labels. As a quick sketch:

- The action should wait until the PR title has been validated (https://github.com/Lilypad-Tech/lilypad/blob/main/.github/workflows/validate_pr_title.yml)
- Get the PR title and parse out the task (e.g. `fix`, `feat`, `chore`, etc)
- Append to the labels with a conventional commit flavored label. This part will likely require querying existing labels and updating with existing labels and the new label.

The conventional commit task should be expanded to full words. For example, a `feat` PR would result in a `feature` label. Our desired mapping of task to label looks like:

https://github.com/Lilypad-Tech/lilypad/blob/b3201ab9e7c016454edcca04cd6b39e870fa5d16/.github/workflows/validate_pr_title.yml#L16-L27

Querying and updating the PR can likely be accomplished using the GitHub CLI within the GitHub action.

---

Co-authored by @bgins

**Comment by rhochmayr** [comment]
Version 1.2.0 seems to deal with this OK. Have you tried that yet?

```
Run ytanikin/PRConventionalCommits@1.2.0
Current labels: [ 'enhancement', 'feature' ]
Task types: ["fix","feat","refactor","perf","spike","hotfix","revert","chore","docs","test","build"]
Custom labels: {
  fix: 'fix',
  feat: 'feature',
  refactor: 'refactor',
  perf: 'performance',
  spike: 'spike',
  hotfix: 'hotfix',
  revert: 'revert',
  chore: 'chore',
  docs: 'documentation',
  test: 'test',
  build: 'build'
}
Managed labels: [
  'fix',           'feat',
  'refactor',      'perf',
  'spike',         'hotfix',
  'revert',        'chore',
  'docs',          'test',
  'build',         'breaking change',
  'feature',       'performance',
  'documentation'
]
New labels: [ 'feature' ]
Current labels: [ 'enhancement', 'feature' ]
New labels: [ 'feature' ]
```

**Comment by bgins** [comment]
We did try `v1.2.0`, but it still removed existing labels in our repo.

Our test case and the place where we need to preserve labels are release PRs created by `release-please`. For example https://github.com/Lilypad-Tech/lilypad/pull/257. `release-please` relies on autorelease labels to determine when to create a release or update a release PR. It's possible that PRConventionalCommits and `release-please` don't interact well, though we haven't spent any time investigating it.

**Comment by rhochmayr** [comment]
Thanks for the info.

Do you mean like this?
![image](https://github.com/user-attachments/assets/7b7d6ae6-8e2c-4538-92f5-aa53e754996c)

Or do you want the labels from the originating PR to be transferred to the release-please PR?

**Comment by bgins** [comment]
> Do you mean like this?

Yeah exactly!

> Or do you want the labels from the originating PR to be transferred to the release-please PR?

No need to transfer labels between PRs. The label only needs to match the PR that it is attached to.

**Comment by rhochmayr** [comment]
Then this, just with version 1.2.0 seems to work.
https://github.com/Lilypad-Tech/lilypad/commit/b3201ab9e7c016454edcca04cd6b39e870fa5d16

The screenshot is from here, where I tried to replicate the setup, just without the publish-binaries steps.
https://github.com/rhochmayr/pr-labeling

**Comment by bgins** [comment]
Nice! That's great. We must not have tested this properly while we were developing the release pipeline. 

Would you be up for putting in a PR to update the action with that? 

As a last check, I will run it in a test repo in the `Lilypad-Tech` organization with settings matching this repo. Heads up that I may not get to it until Monday.

**Comment by bgins** [comment]
Closed by #260
